cmd/http-service/server/backup.go

```diff
@@ -431,11 +461,128 @@ func convertToRestoreInfo(restore *v1alpha1.Restore) *api.RestoreInfo {
+	kubeCli := s.KubeClient.GetKubeClient(k8sID)
+	logger := log.L().With(zap.String("request", "StopBackup"), zap.String("k8sID", k8sID),
+		zap.String("clusterID", req.ClusterId), zap.String("backupID", req.BackupId))
+	if opCli == nil || kubeCli == nil {
+		logger.Error("K8s client not found")
+		message := fmt.Sprintf("no %s is specified in the request header or the kubeconfig context not exists", HeaderKeyKubernetesID)
+		setResponseStatusCodes(ctx, http.StatusBadRequest)
+		return &api.StopBackupResp{Success: false, Message: &message}, nil
+	}
+
+	// check whether the backup exists
+	backup, err := opCli.PingcapV1alpha1().Backups(req.ClusterId).Get(ctx, req.BackupId, metav1.GetOptions{})
+	if err != nil {
+		logger.Error("Backup not found", zap.Error(err))
+		message := fmt.Sprintf("Backup %s not found", req.BackupId)
+		setResponseStatusCodes(ctx, http.StatusBadRequest)
+		return &api.StopBackupResp{Success: false, Message: &message}, nil
+	}
+
+	// stop backup
+	if backup.Spec.Mode == v1alpha1.BackupModeLog {
+		backup.Spec.LogStop = true
+		_, err = opCli.PingcapV1alpha1().Backups(req.ClusterId).Update(ctx, backup, metav1.UpdateOptions{})
+		if err != nil {
+			logger.Error("Stop log backup failed", zap.Error(err))
+			message := fmt.Sprintf("Stop log backup failed: %s", err.Error())
+			setResponseStatusCodes(ctx, http.StatusInternalServerError)
+			return &api.StopBackupResp{Success: false, Message: &message}, nil
+		}
+	} else {
+		_, err := kubeCli.BatchV1().Jobs(backup.GetNamespace()).Get(ctx, backup.GetBackupJobName(), metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logger.Warn("Backup is already Stopped", zap.Error(err))
+				message := fmt.Sprintf("Backup %s is already Stopped", req.BackupId)
+				setResponseStatusCodes(ctx, http.StatusNotFound)
+				return &api.StopBackupResp{Success: false, Message: &message}, nil
+			}
+			logger.Error("Get backup job failed", zap.Error(err))
+			message := fmt.Sprintf("Get backup job failed: %s", err.Error())
+			setResponseStatusCodes(ctx, http.StatusInternalServerError)
+			return &api.StopBackupResp{Success: false, Message: &message}, nil
+		}
+
+		err = kubeCli.BatchV1().Jobs(backup.GetNamespace()).Delete(ctx, backup.GetBackupJobName(), metav1.DeleteOptions{})
+		if err != nil {
+			logger.Error("Stop backup failed", zap.Error(err))
+			message := fmt.Sprintf("Stop backup failed: %s", err.Error())
+			setResponseStatusCodes(ctx, http.StatusInternalServerError)
+			return &api.StopBackupResp{Success: false, Message: &message}, nil
+		}
+	}
+
+	// update backup status
+	_, err = opCli.PingcapV1alpha1().Backups(req.ClusterId).Update(ctx, backup, metav1.UpdateOptions{})
+	if err != nil {
+		logger.Error("Backup not found", zap.Error(err))
+		message := fmt.Sprintf("Backup %s not found", req.BackupId)
+		setResponseStatusCodes(ctx, http.StatusBadRequest)
+		return &api.StopBackupResp{Success: false, Message: &message}, nil
+	}
```

Comment:
Can we remove these lines? and also for `// update restore status`

Reply:
removed